src/intrinsics.cpp

```diff
@@ -918,6 +920,180 @@ static Value *emit_intrinsic(intrinsic f, jl_value_t **args, size_t nargs,
     Value *den;
     Value *typemin;
     switch (f) {
+    HANDLE(bytevec_ref,2) {
```

Comment:
I'm fine with an intrinsic for now. I really don't want to add more "builtins", and it would be best to gradually remove them if possible.

@vtjnash excellent code review, thanks.